docs/src/main/paradox/common/http-model.md

```diff
@@ -108,6 +121,19 @@ In addition to the simple @scala[@apidoc[HttpEntity] constructors]@java[`HttpEnt
 as shown here the Akka HTTP model defines a number of subclasses of @apidoc[HttpEntity] which allow body data to be specified as a
 stream of bytes. @java[All of these types can be created using the method on `HttpEntites`.]
 
+@@@ note { title='String representation' }
+
+There are certain environments where it is easy to inadvertently print, write or log entries built out of string representations of @apidoc[HttpResponse] instances. On the other hand, it is not uncommon for HTTP headers and entities to contain _Personal Identifying Information (PII)_ or _Sensitive Personal Information (SPI)_ . 
+
+To avoid accidentally leaking such information, these fields are omitted from @apidoc[HttpResponse] `toString` output. 
+
+If needed, it is possible to alter this behavior using type classes as shown in the following examples: 
+
+Scala
+:   @@snip [HttpResponseShow.scala]($test$/scala/docs/http/scaladsl/HttpResponseShow.scala) 
+
+@@@
```

Comment:
Missing corresponding sample how to handle it for Java. I think for Java it might be less probable that type class is a known concept, so perhaps a a separate phrasing there as well.

Actually, to be honest, I think just having a method (like `def toDetailedString(res): String`) would suffice for Scala as well, people who wants to use type classes will know how to do that anyway.

Reply:
Thanks @johanandren ! Are you suggesting adding this behaviour:

```scala
s"""HttpRequest(${_1},${_2},${_3},${_4},${_5})"""
```

as a separated `def toDetailedString(res): String` method, living along with `toString` ?

Reply:
That's perfectly fine for me, I can add it and update the docs if that approach is OK.

Reply:
Or are you suggesting that the doc should show just an example of such a method taking a response or a request as its input?

Reply:
Just a regular method or function going from HTTPRequest to the desired string format rather than a typeclass and implicit decoration. Would also make the Java and Scala samples pretty much identical.

Reply:
Done, please check 02212f21b8df759d465f94235f15f5b6d07c8ef7 .

Reply:
Great, thanks!